Documentation/DependencyInjection.md

```diff
@@ -0,0 +1,77 @@
+# Dependency Injection
+Shortcut Foundation provides a convenient way to access the same object anywhere in your project via dependency injection.
+
+## Usage
+To inject an object, e.g. an instance of CameraManager, use the @Inject property wrapper
+```
+@Inject var cameraManager: CameraManager
+```
+and then map it to its Type in the configure(_ injector:) method of a Config.
+```
+struct AppConfig: Config {
+    func configure(_ injector: Injector) {
+        injector.map(CameraManager.self) {
+            CameraManager()
+        }
+    }
+}
+```
+An instance of Config is then used to initialize a Context instance on app launch, which is held for the lifecycle of the app.
+```
+@main
+struct MyApp: App {
+    let context = Context(AppConfig())
+
+    var body: some Scene {
+        WindowGroup {
+            RootView()
+        }
+    }
+}
+```
+
+That's it!
+
+### Important!
+When mapping objects in configure(_ injector:), the order of the mapping is important.
+
+If mapping an instance of ViewModel, which refers to an instance of SubviewModel, then ViewModel must be mapped before SubviewModel.
+```
+struct AppConfig: Config {
+    func configure(_ injector: Injector) {
+        injector.map(ViewModel.self) {
+            ViewModel()
+        }
+        injector.map(SubviewModel.self) {
+            SubviewModel()
+        }
+    }
+}
+```
+If not mapped in the correct order, the app may crash.
+
+### Mapping Protocols
+In some situations it can be useful to map the type of a protocol instead of a class. By then having several classes conform to the protocol, you can add logic to load the correct instance where appropriate.
+
+Let's say you have an ApiService class that handles all your API calls when running your app, and you have a MockApiService class that you want to use for testing. By having both ApiService and MockApiService conform to e.g. IApiService, you can map
+```
+injector.map(IApiService.self) {
+    isRunningTests ? MockApiService() : ApiService() as IApiService
+}
+```
+#### Example Logic
+In the above example, isRunningTests is defined as
+```
+var isRunningTests: Bool { ProcessInfo.processInfo.environment["XCTestConfigurationFilePath"] != nil }
+```
+Another situation where this approach could be useful is for SwiftUI previews. Using the same approach as above, we can define e.g. isRunningInPreview as
+```
+var isRunningInPreview: Bool { ProcessInfo.processInfo.environment["XCODE_RUNNING_FOR_PREVIEWS"] == "1" }
+```
+
+## Variations
+Along with the standard @Inject, there's an additional four variations of the property wrapper, which can be used in its place when needed.
```

Comment:
A bit grammatical correction: instead of `there's` you can say **there are additional four**.

Reply:
Fixed in [f63a4a7](https://github.com/shortcut/shortcut-foundation-ios/pull/49/commits/f63a4a771092691251119fa8b23c95b7c6d94d23)